src/main/java/org/eclipse/golo/runtime/AugmentationMethodFinder.java

```diff
@@ -10,282 +10,72 @@
 package org.eclipse.golo.runtime;
 
 import java.lang.invoke.*;
-import java.lang.reflect.Method;
-import java.util.List;
-import static java.util.stream.Collectors.toList;
-import static java.util.Arrays.asList;
+import java.util.stream.Stream;
+import java.util.Comparator;
+import java.util.Objects;
+import org.eclipse.golo.runtime.augmentation.DefiningModule;
 
 import static java.lang.invoke.MethodHandles.*;
 import static java.lang.reflect.Modifier.*;
-import static org.eclipse.golo.runtime.DecoratorsHelper.getDecoratedMethodHandle;
-import static org.eclipse.golo.runtime.DecoratorsHelper.isMethodDecorated;
 
-class AugmentationMethodFinder implements MethodFinder {
```

Comment:
Ok, so `MethodInvocation` is a domain object to describe a particular lookup. Fine refactoring :smile: